include/static-pages.php

```diff
@@ -122,23 +121,35 @@ public function page_link( $link, $id ) {
 	}
 
 	/**
-	 * Adds page_on_front and page_for_posts properties to the language objects.
+	 * Adds page_on_front and page_for_posts properties to language data before the object is created.
 	 *
-	 * @since 1.8
+	 * @since 3.4
+	 *
+	 * @param array $static_pages Array of language page_on_front and page_for_posts properties.
+	 * @param array $language Language data.
+	 * @return array {
+	 *     Static pages properties if exist.
 	 *
-	 * @param PLL_Language[] $languages The list of languages.
-	 * @param PLL_Model      $model     The instance of PLL_Model.
-	 * @return PLL_Language[]
+	 *     @type int $page_on_front  ID of the page on front in this language
+	 *     @type int $page_for_posts Id of the page for posts in this language
+	 * }
 	 */
-	public static function pll_languages_list( $languages, $model ) {
+	public function set_static_pages( $static_pages, $language ) {
 		if ( 'page' === get_option( 'show_on_front' ) ) {
-			foreach ( $languages as $k => $language ) {
-				$languages[ $k ]->page_on_front = $model->post->get( get_option( 'page_on_front' ), $language );
-				$languages[ $k ]->page_for_posts = $model->post->get( get_option( 'page_for_posts' ), $language );
+			$this->page_on_front  = intval( get_option( 'page_on_front' ) );
```

Comment:
Not sure `if ( 'page' === get_option( 'show_on_front' )` is useful here, because it's already checked before adding the hook in `$this->init()`.

Reply:
Sure! You're right. Good catch 👍Thanks.
Done in 7b5b9532109ce394317e1ebe3e088239e9727bd0